In [1]:
# Import necessary Python libraries.
import pandas as pd
from pathlib import Path

In [2]:
# Make connection to the source file(s).
miss_csv = Path(r"C:\Users\admin\Desktop\FinTech\FinTech2022_Module3_Financial Analysis_Pandas\01-Missing_Data\items.csv")


In [3]:
# Make Pandas read the CSV file.
# Creat a dataframe to manipulate.
miss_df = pd.read_csv(miss_csv)

In [4]:
# Make the dataframe to display first and last five itmes.
miss_df.head()

# C.P.A. -> there is a missing value.

Item Number  Price   Quantity
0         2468    2.99       250
1         1357     NaN       300
2         9753    3.47       350

In [5]:
# Since there is a missing value/data, manipulating it is a requirement.
# Transform dataframe to return True or False.
miss_df.isnull()

Item Number  Price   Quantity
0        False   False     False
1        False    True     False
2        False   False     False

In [6]:
# Take a total sum of the missing data/values.
miss_df.isnull().sum()

 

Item Number    0
Price          1
Quantity       0
dtype: int64

# C.P.A. 
## Materially: Significant or Non-significant
**Materiality** of the amount of missing data/values in the context   
of acquisitions, reporting, etc.
#### Statistics
Applying statistic can be a determining factor.

In [7]:
# Applying statistics
miss_df.isnull().mean()

Item Number    0.000000
Price          0.333333
Quantity       0.000000
dtype: float64

# Conclusion of missing data:
Missing data is __material__ making up 33.33% of its entirety.  
### What to do next?

# From the class PowerPoint are the following items for exercise.

In [8]:
import pandas as pd
import numpy as np

apple_df = pd.DataFrame({"AAPL": [72.27, np.nan, 74.39, np.nan, 75.94, 76.93,np.nan, 78.74, np.nan, 79.81, np.nan, 79.53, np.nan, 79.56, 79.49]})

apple_df

AAPL
0   72.27
1     NaN
2   74.39
3     NaN
4   75.94
5   76.93
6     NaN
7   78.74
8     NaN
9   79.81
10    NaN
11  79.53
12    NaN
13  79.56
14  79.49

In [9]:
# There are a few missing datas/values.
# Requrements: manipulate missing items
apple_df.isnull()

AAPL
0   False
1    True
2   False
3    True
4   False
5   False
6    True
7   False
8    True
9   False
10   True
11  False
12   True
13  False
14  False

In [10]:
# Sum statistics about the missing items.
apple_df.isnull().sum()


AAPL    6
dtype: int64

In [11]:
# Average statistics about the missing items.
apple_df.isnull().mean()

AAPL    0.4
dtype: float64

# Conclusion about the number of missing items:
Missing data/values are __material__ making up 40% of its entirety.
## What to do next?